### PR TITLE
feat: Add pinterest

### LINKED ISSUE #249: [social-media/publish-post] Add Pinterest
Another popular provider requested by existing user for the social-media profiles would be Pinterest, primarily publishing a "pin" to a "board".

Links:

- [Pinterest REST API](https://developers.pinterest.com/docs/api/v5/)
- [Content apps overview](https://developers.pinterest.com/docs/solutions/content-apps/)

This would need the following maps:

- `social-media/publishing-profiles` returns the list of authorized users' boards
  - https://developers.pinterest.com/docs/api/v5/#operation/boards/list
- `social-media/publish-post` creates an image pin
  - https://developers.pinterest.com/docs/api/v5/#operation/pins/create
  - image can be passed with the request as base64 encoded string or `image_url`, so we can support both options passed from the profile

In addition we could 

Notes:
- For initial auth flow, use probably [passport-pinterest-v3](https://github.com/mystery-man/passport-pinterest-v3), which seems to be the least outdated one
- Boards can be further divided into sections, which can be (optionally) published into
  - I'd omit this from the first version of the map. If we decide to add this, we can expose sections from `publishing-profiles` as a flat output and serialize their ID with the board ID (eg. `{boardID}/{sectionID}`).

### PR DESCRIPTION
**Pull request by jnv**
## Description

Adds Pinterest maps for the following profiles:

- social-media/publishing-profiles
- social-media/publish-post
- oauth2/refresh-token

## Motivation and Context

Fixes #249.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: grid/social-media/publish-post/maps/pinterest.suma [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  set if (media.url) {
+    mediaSource = {
+      source_type: 'image_url',
+      url: media.url,
+    }
+  }
+
+  // return map error if (!mediaSource.url && !mediaSource.content_type) {
```
Comment (janhalama) [inline grid/social-media/publish-post/maps/pinterest.suma:43]: Why is this commented out? Does Pinterest accept media without defined `content_type`?

File: nock/communication/send-email/mailchimp/SendEmail/recording-69d92e19523ced971d75fcf53bc8882e.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
     ],
     "rawHeaders": [
       "server",
-      "nginx/1.12.2",
+      "nginx/1.4.6 (Ubuntu)",
```
Comment (janhalama) [inline nock/communication/send-email/mailchimp/SendEmail/recording-69d92e19523ced971d75fcf53bc8882e.json:38]: Is this intended change? Looks like it was part of different PR.

File: nock/communication/send-email/mailchimp/SendEmail/recording-a10433b7962d1f96e4eee9be0141d6dd.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
     ],
     "rawHeaders": [
       "server",
-      "nginx/1.12.2",
+      "nginx/1.4.6 (Ubuntu)",
```
Comment (janhalama) [inline nock/communication/send-email/mailchimp/SendEmail/recording-a10433b7962d1f96e4eee9be0141d6dd.json:25]: DITTO

**Review by janhalama — APPROVED** [review]
LGTM, I have few minor comments.

File: nock/communication/send-email/mailchimp/SendEmail/recording-69d92e19523ced971d75fcf53bc8882e.json (hunk 2 above)
Comment (jnv) [reply nock/communication/send-email/mailchimp/SendEmail/recording-69d92e19523ced971d75fcf53bc8882e.json:38]: You are right, this shouldn't be there. Good catch!

File: providers/pinterest.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "pinterest",
+  "services": [
+    {
+      "id": "api",
+      "baseUrl": "https://api.pinterest.com"
+    },
+    {
```
Comment (tmladek) [inline providers/pinterest.json:8]: what's this service for?

**Review by tmladek — APPROVED** [review]

File: providers/pinterest.json (hunk 4 above)
Comment (tmladek) [reply providers/pinterest.json:8]: (the `www` one; github's inline view is weird)

File: grid/social-media/publish-post/maps/pinterest.suma (hunk 1 above)
Comment (jnv) [reply grid/social-media/publish-post/maps/pinterest.suma]: Good point, I have omitted this part with intention to test it first, but forgot about it.

**Comment by jnv** [comment]
@janhalama I have added somewhat extensive test for media upload, so please take another look.

File: providers/pinterest.json (hunk 4 above)
Comment (jnv) [reply providers/pinterest.json:8]: For `https://www.pinterest.com/oauth/`: https://developers.pinterest.com/docs/api/v5/#section/Start-the-OAuth-flow

File: grid/oauth2/refresh-token/maps/refresh-token.ts [hunk 5]
```diff
@@ -87,11 +87,12 @@ export const refreshTokenTest = (
...
-      const { accessToken, refreshToken, ...rest } = resultUnwrapped;
-      expect(accessToken).toEqual(expect.any(String));
-      expect(rest).toMatchSnapshot();
+      const { refreshToken, ...rest } = resultUnwrapped;
+      expect(rest).toMatchSnapshot({
+        accessToken: expect.any(String),
+      });
       if (refreshToken) {
-        currentInput.refreshToken = refreshToken;
+        currentInput.refreshToken = resultUnwrapped.refreshToken;
```
[the remark this answers is not in the document]
Comment (jnv) [reply grid/oauth2/refresh-token/maps/refresh-token.ts:95]: Good point, I was messing around and forgot about it.

File: grid/social-media/publish-post/maps/publish-media-post.ts [hunk 6]
```diff
@@ -31,32 +46,78 @@ export const publishMediaPostTest = (
...
-                    },
-                  ],
-                },
+          const profiles = await getPublishingProfiles(provider);
+
+          const result = await superfacePublisPost.run(
+            {
+              useCase: 'PublishPost',
+              input: {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any
```
Comment (janhalama) [inline grid/social-media/publish-post/maps/publish-media-post.ts:55]: This line is no more relevant.
```suggestion
```

**Review by janhalama — APPROVED** [review]
Looks good. I like the way your refactored publishing media tests 👍 

I have just one minor comment.

**Pull request merged**